src/algebra/parity.lean

```diff
@@ -241,6 +241,21 @@ lemma odd.neg_one_pow (h : odd n) : (-1 : α) ^ n = -1 := by rw [h.neg_pow, one_
 
 end monoid
 
+section canonically_ordered_comm_semiring
+
+variables [canonically_ordered_comm_semiring α]
+
+-- TODO: this holds more generally in a `canonically_ordered_add_monoid` if we refactor `odd` to use
+-- either `2 • t` or `t + t` instead of `2 * t`.
+lemma odd.pos  [nontrivial α] : ∀ {n : α}, odd n → 0 < n
+| _ ⟨k, rfl⟩ :=
```

Comment:
Seems weird to use the equation compiler here when there's no inductive data types involved. I recommend you rewrite this using `rintros`.

Reply:
sure 🤷

Reply:
Equation compiler is basically term mode rintros, it can't always be used in the middle of a proof but this style is fairly common in low level files.

Reply:
I agree that this style is common and not bad.